### ISSUE TITLE
[macOS/Darwin] pthread_mutex_lock: Invalid argument

### ISSUE THREAD

**Issue by jonthn** [nanomsg/nng#1417]
**Describe the bug**

Only on some macOS systems it will trigger this issue of `pthread_mutex_lock` with a message in the console/terminal to make a bug report. It's not clear when this happens but it's a normal use of `nng` and it works normally for most macOS system but some will quickly trigger the issue.

**Expected behavior**

It should work normally without crash/panic.

**Actual Behavior**

It panics

```
(lldb) bt
* thread #2, stop reason = signal SIGABRT
  * frame #0: 0x00007fff6df7fb66 libsystem_kernel.dylib`__pthread_kill + 10
    frame #1: 0x00007fff6e14a080 libsystem_pthread.dylib`pthread_kill + 333
    frame #2: 0x00007fff6dedb1ae libsystem_c.dylib`abort + 127
    frame #3: 0x000000010d4e99c9 libnng.1.3.dylib`nni_plat_abort + 9
    frame #4: 0x000000010d4dddb1 libnng.1.3.dylib`nni_panic + 529
    frame #5: 0x000000010d4f19ca libnng.1.3.dylib`nni_pthread_mtx_lock + 58
    frame #6: 0x000000010d4f1985 libnng.1.3.dylib`nni_plat_mtx_lock + 21
    frame #7: 0x000000010d4e6a25 libnng.1.3.dylib`nni_mtx_lock + 21
    frame #8: 0x000000010d4f3a5d libnng.1.3.dylib`nni_posix_poll_thr + 477
    frame #9: 0x000000010d4e6d58 libnng.1.3.dylib`nni_thr_wrap + 184
    frame #10: 0x000000010d4f1e18 libnng.1.3.dylib`nni_plat_thr_main + 120
    frame #11: 0x00007fff6e147661 libsystem_pthread.dylib`_pthread_body + 340
    frame #12: 0x00007fff6e14750d libsystem_pthread.dylib`_pthread_start + 377
    frame #13: 0x00007fff6e146bf9 libsystem_pthread.dylib`thread_start + 13
```

**To Reproduce**

I'm not able to produce a way to make this, as it's already not easy to make it happens, for example I'm not able to trigger the issue on my development computer.

**Environment Details**

 - latest nng version `1.3.2`
 - macOS 'High Sierra' 10.13.6
 - clang (`clang version 8.0.0 (tags/RELEASE_800/final 360744)`, with C code
 - Shared library

**Additional context**

CMake options (using mbedTLS 2.16.0):

```
-DCMAKE_BUILD_TYPE=Release\
-DNNG_ENABLE_TLS=ON\
-DMBEDTLS_ROOT_DIR=/my/mbedTLS-dist/\
-DBUILD_SHARED_LIBS=ON\
-DNNG_ENABLE_STATS=ON\
```

**Comment by gdamore** [comment]
Its very strange that this would happen under normal circumstances.  Is the socket being closed?  Or are pipes being closed?  I'm wondering if something is in transition when this occurs.

**Comment by gdamore** [comment]
Can you test your application with master?  I've made a lot of (unreleased) changes since 1.3.2.

**Comment by gdamore** [comment]
I'm particularly keen to know if the fix for #1275 has changed this behavior at all.

**Comment by jonthn** [comment]
It seems it fixed the issue. But I would not be able to tell which commit.

I will update to release v1.4.0 and if it appears again I will either re-open or create a new issue.

Thank you.
